[3809] Content Tree Panel

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/3809

aachavarria committed:
adding ContentTreePanel;

aachavarria committed:
fixes

aachavarria committed:
adding contentTree icons;
contentTree changes;

aachavarria committed:
adding support to open components as root

aachavarria committed:
adding previous support

aachavarria committed:
adding expanded to components, and working on scroll TO

aachavarria committed:
adding scroll and pulse animation,
fixing contentTree ids

aachavarria committed:
fixing to avoid open the tree on labelClick;
scrollToElement only on labelClick;

aachavarria committed:
fixes to the scroll;
pending scroll to repeat items childs;

aachavarria committed:
Merge branch '2019-11-ui-restructure' of https://github.com/craftercms/studio-ui into feature/3809

 Conflicts:
	ui/app/src/state/actions/preview.ts
	ui/guest/src/util.js

aachavarria committed:
fixing merge issue

aachavarria committed:
Fixes on scrollToElement, and repeatGroupChildren

aachavarria committed:
contentTree Changes to TreeItemCustom<component>

aachavarria committed:
Adding ComponentMenu

rart commented on ui/app/src/components/Icons/Field.tsx:
```diff
@@ -0,0 +1,29 @@
+/*
```
Rename to ContentTypeField

rart commented on ui/app/src/components/Icons/Repeat.tsx:
```diff
@@ -0,0 +1,27 @@
+/*
```
Repeat => RepeatGroup

rart commented on ui/app/src/modules/Preview/ToolBar.tsx:
```diff
@@ -290,80 +192,11 @@ export function AddressBar(props: AddressBarProps) {
-      }
+        setAnchorEl={setAnchorEl}
+        site={site}
+      />
```
- Why does the ComponentMenu need the ability to **set** the anchor?
- Follow same API as material components (use both anchor and open props, onClose, etc)

rart commented on ui/app/src/modules/Preview/Tools/ContentTree.tsx:
```diff
@@ -0,0 +1,400 @@
+import TreeView from '@material-ui/lab/TreeView';
+import IconButton from '@material-ui/core/IconButton';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
```
use rounded set

rart commented on ui/app/src/modules/Preview/Tools/ContentTree.tsx:
```diff
@@ -0,0 +1,400 @@
+import clsx from 'clsx';
+import { defineMessages, useIntl } from 'react-intl';
+import ToolPanel from './ToolPanel';
+import { createStyles, makeStyles } from '@material-ui/core';
```
import from `/style`. always avoid importing from `/core`

rart commented on ui/app/src/modules/Preview/Tools/ContentTree.tsx:
```diff
@@ -0,0 +1,400 @@
+  });
+
+  useEffect(() => {
+    if (guest?.modelId && guest?.models && contentTypesBranch.byId && data.selected === null) {
```
once you've done `guest?.`, on the subsequent uses is not necessary to use the opt chaining op as the first if would have checked for _nullish-ness_

rart commented on ui/app/src/state/actions/preview.ts:
```diff
@@ -64,6 +64,7 @@ export const CONTENT_TYPE_RECEPTACLES_REQUEST = 'CONTENT_TYPE_RECEPTACLES_REQUES
 export const CONTENT_TYPE_RECEPTACLES_RESPONSE = 'CONTENT_TYPE_RECEPTACLES_RESPONSE';
 export const SCROLL_TO_RECEPTACLE = 'SCROLL_TO_RECEPTACLE';
 export const CLEAR_HIGHLIGHTED_RECEPTACLES = 'CLEAR_HIGHLIGHTED_RECEPTACLES';
+export const SCROLL_TO_ELEMENT = 'SCROLL_TO_ELEMENT';
```
Since we're also animating, etc use a more specific and semantic action type: CONTENT_TREE_FIELD_SELECTED

rart commented on ui/app/src/styles/icon.ts:
```diff
@@ -0,0 +1,22 @@
+  stroke: 'currentColor'
+};
+
+export default iconStyles;
```
1. rename const to `iconWithStrokeAndFill`
2. export const
3. don't export default

rart commented on ui/guest/src/classes/ContentController.js:
```diff
@@ -821,6 +821,12 @@ function fetchById(id, site = Cookies.get('crafterSite')) {
+                    namebonito_s
+                    namefeo_s
+                  }
+                }
```
Rollback. Don't commit this.

rart commented on ui/guest/src/components/Guest.js:
```diff
@@ -16,6 +16,8 @@
 
 import React, { useEffect, useRef, useState } from 'react';
+import jss from 'jss';
+import preset from 'jss-preset-default';
```
x

rart commented on ui/guest/src/components/Guest.js:
```diff
@@ -70,6 +73,27 @@ const escape$ = fromEvent(document, 'keydown').pipe(
+  }
+};
+
+const {classes} = jss.createStyleSheet(styles).attach()
```
All these styles go to the styles file. Guest would only apply the class.

rart commented on ui/guest/src/components/Guest.js:
```diff
@@ -45,6 +47,7 @@ import {
   notNullOrUndefined,
   pluckProps,
   RELOAD_REQUEST,
+  SCROLL_TO_ELEMENT,
```
Rename as requested above.

rart commented on ui/guest/src/components/Guest.js:
```diff
@@ -1142,6 +1166,33 @@ export function Guest(props) {
+        pulseAnimation($element);
+      }
+    }
+  }
```
Looks like all these functions can be outside of the component. Move out all that can be moved out - including others below and above not part of this change set.

rart commented on ui/guest/src/util.js:
```diff
@@ -67,6 +67,7 @@ export const CONTENT_TYPE_RECEPTACLES_REQUEST = 'CONTENT_TYPE_RECEPTACLES_REQUES
 export const CONTENT_TYPE_RECEPTACLES_RESPONSE = 'CONTENT_TYPE_RECEPTACLES_RESPONSE';
 export const SCROLL_TO_RECEPTACLE = 'SCROLL_TO_RECEPTACLE';
 export const CLEAR_HIGHLIGHTED_RECEPTACLES = 'CLEAR_HIGHLIGHTED_RECEPTACLES';
+export const SCROLL_TO_ELEMENT = 'SCROLL_TO_ELEMENT';
```
Rename as requested above.

aachavarria committed:
template-editor.js: conditional on pendingChanges;
componentMenu: adding interfaces; changes to receive modelId;
contentTree: adding ComponentMenu to contentTree;
toolbar: sending modelId;

aachavarria committed:
ComponentMenu: adding embeddedParentPath and parentId; to being used on legacyForm, and publish dialog;
PublishDialog: adding support to OnSuccess, adding validations;
contentTree: adding embeddedParentPath to Component Item;
toolbar: moving handle close to top level;

aachavarria committed:
ComponentMenu: validations on Controller Option and Tab;
EmbeddedLegacyEditors: adding EmbeddedLegacyEditorsProps;

aachavarria committed:
adding anchorOrigin to ComponentMenu props;
ContentTree: fixes on anchorEl;
Guest: Validations fix on pulseAnimation; scrollToElement;

aachavarria committed:
adding missing validation;

rart commented on static-assets/components/cstudio-forms/template-editor.js:
```diff
@@ -274,7 +274,9 @@ CStudioAuthoring.Module.requireModule(
                       aceEditor.getSession().on('change', function() {
                         aceEditor.isModified = true;
-                        onSaveCb.pendingChanges();
+                        if (onSaveCb.pendingChanges()) {
```
Did you mean to check for the nullishness or pending changes instead of calling it?

aachavarria replied:
nullishness

rart commented on ui/app/src/components/ComponentMenu.tsx:
```diff
@@ -0,0 +1,235 @@
+          setPublishDialog({ item })
+        },
+        (error) => {
+          console.log(error)
```
Handle properly.

aachavarria committed:
ContentController: revert File;
Util.js, Guest.js: adding new utils;
styles.js, Guest.js: moving animation pulse to styles;
General: renaming files and variables;

rart commented on ui/app/src/components/ComponentMenu.tsx:
```diff
@@ -0,0 +1,235 @@
+      }
+      {
+        publishDialog.open &&
+        <PublishDialog
+          scheduling={publishDialog.scheduling}
+          items={[publishDialog.item]}
+          onSuccess={onSuccessPublish}
+          onClose={onClosePublish}
+        />
```
We could set these higher above in the component tree so less components would require this code within them.

aachavarria replied:
pending

rart commented on ui/app/src/modules/Content/Publish/PublishDialog.tsx:
```diff
@@ -180,7 +183,7 @@ function PublishDialog(props: PublishDialogProps) {
 
     // call externalClose fn
-    onClose();
+    onClose && onClose();
```
`onClose?.()`

rart commented on ui/app/src/modules/Content/Publish/PublishDialog.tsx:
```diff
@@ -203,7 +206,8 @@ function PublishDialog(props: PublishDialogProps) {
         setApiState({ ...apiState, error: false, submitting: false });
-        onClose(response);
+        onSuccess && onSuccess(response);
+        onClose && onClose(response);
```
`onSuccess?.()`
`onClose?.()`

rart commented on ui/app/src/modules/Preview/Tools/ContentTree.tsx:
```diff
@@ -0,0 +1,450 @@
+            <IconButton className={classes.options} onMouseOver={(e) => setOverState(e, true)}
+                        onClick={(e) => handleOptions(e, nodes.modelId, nodes.parentId, nodes.embeddedParentPath)}>
+              <MoreVertIcon/>
+            </IconButton>
```
Format

aachavarria committed:
PR Changes

aachavarria committed:
Format

aachavarria committed:
adding ErrorDialog;
Adding error handling to componentMenu;

rart committed:
Update ContentController.js

aachavarria committed:
Merge branch '2019-11-ui-restructure' of https://github.com/craftercms/studio-ui into feature/3809

 Conflicts:
	ui/guest/src/components/Guest.tsx
	ui/guest/src/util.tsx

aachavarria committed:
adding getElementRegistry;
adding RenderTree model;
adding scrollToReceptacle, scrollToNode to utils.tsx;
